### ISSUE TITLE
requirements_test can take long time (`>60s`) due to `--no-cache` usage

### ISSUE THREAD

**Issue by shayanhoshyari** [theoremlp/rules_uv#247]
I have an input requirements that include `torch`

```
# requirements.txt 
click
jinja2
licenseheaders
mypy
pydantic
pytest
ruamel.yaml
sphinx
torch==2.7.0
```

Since torch has very different set of deps on linux and macos, I compile this to two requirements

```py
pip_compile(
    name = "requirements_linux",
    python_platform = "x86_64-manylinux_2_28",
    requirements_in = "pyproject.toml",
    requirements_txt = "requirements.linux.txt",
    timeout = "moderate", # Without this, it is consistently timing out
)

pip_compile(
    name = "requirements_macos",
    python_platform = "aarch64-apple-darwin",
    requirements_in = "pyproject.toml",
    requirements_txt = "requirements.macos.txt",
)

multirun(
    name = "requirements.update",
    commands = [
        ":requirements_linux",
        ":requirements_macos",
    ],
    jobs = 1,
)
```

After a first run, `requirements.update` runs in a reasonable amount of time (5-10 secs). However, `requirements_linux_test` consistently takes `> 60 s`.

Note that torch on linux contains more than 4 GB worth of wheels.

requirements_test is hardcoded to use [`--no-cache`](https://github.com/theoremlp/rules_uv/blob/v0.69.0/uv/private/pip_compile_test.sh#L17), I suspect that is at fault.

**Comment by baxelrod-bdai** [comment]
Does this flag help?  
```
bazel test //requirements_linux_test --test_timeout=300
```

**Comment by shayanhoshyari** [comment]
> Does this flag help?
> 
> ```
> bazel test //requirements_linux_test --test_timeout=300
> ```

What we do in CI and developer workflows is simply running `bazel test ...`.  And one of the attractions was how simple it is. Same command local and CI, and it only does extra work when it detects changes.

I am looking for a solution that can keep things this way.
